vyper/semantics/types/user.py

```diff
@@ -153,6 +153,13 @@ class EventT(_UserType):
         Name of the event.
     """
 
+    _invalid_locations = (
```

Comment:
it will be helpful to either autogenerate this list or have a global constant with all the members of `DataLocation` - in case we add locations in the future, it will prevent bugs.

Reply:
i think `tuple(iter(DataLocation))` will do the trick

Reply:
Do we want to exclude `DataLocation.UNSET`?

Reply:
yea that seems like it was a good call - it was causing test failures right?

Reply:
> yea that seems like it was a good call - it was causing test failures right?

It wasn't for the updated test files and examples directory at least. It does affect which exceptions are raised for invalid instantiations though.

Using this example contract
```
event Foo:
    a: uint256

f: constant(Foo) = Foo(1)
```

The error messages are if DataLocation.UNSET is excluded and included respectively. The latter looks more like the right place to catch the exception, but I think we will need to manually amend the error message.
```
vyper.exceptions.StateAccessViolation: Value must be a literal
  contract "contracts/event.vy:6", line 6:19 
       5
  ---> 6 f: constant(Foo) = Foo(1)
  --------------------------^
       7
```
```
vyper.exceptions.InvalidType: event Foo(uint256) is not instantiable in DataLocation.UNSET
  contract "contracts/event.vy:6", line 6:12 
       5
  ---> 6 f: constant(Foo) = Foo(1)
  -------------------^
       7
```